--- handler.sh-void	2019-09-07 10:33:23.532671912 +0200
+++ handler.sh	2019-09-07 11:21:39.369708027 +0200
@@ -1,6 +1,15 @@
 #!/bin/sh
 # Default acpi script that takes an entry for all actions
 
+# Ignore things if running in a Desktop session
+if type ck-list-sessions ; then
+  x11_displays=$(ck-list-sessions | grep "x11-display-device = '/dev" | wc -l)
+  if [ $x11_displays -gt 0 ] ; then
+    echo "In a X11 session, ignore this..."
+    exit 0
+  fi
+fi
+
 # NOTE: This is a 2.6-centric script.  If you use 2.4.x, you'll have to
 #       modify it to not use /sys
 
