--- handler.sh-void	2019-08-24 22:17:29.000000000 +0200
+++ handler.sh	2020-03-19 23:08:49.753978989 +0100
@@ -1,6 +1,15 @@
 #!/bin/sh
 # Default acpi script that takes an entry for all actions
 
+# Ignore things if running Desktop power-manager
+for mgr in mate-power-manager gsd-power
+do
+  if [ -n "$(pidof $mgr)" ] ; then
+    echo "$mgr is running ... (let it handle ACPI events)"
+    exit 0
+  fi
+done
+
 # NOTE: This is a 2.6-centric script.  If you use 2.4.x, you'll have to
 #       modify it to not use /sys
 
